contracts/Distributor.sol

```diff
@@ -0,0 +1,250 @@
+        (bool success,) = account.call{value: ethClaimable}("");
+        require(success, "Distributor: Unable to receive ETH from contract.");
+
+        emit ETHFundClaimed(account, ethClaimable);
+    }
+
+    function payoutToken(address account_, IERC20 token_) public {
+        address account = account_;
+        IERC20 token = token_;
+        require(_sharesForTokens[token][account] > 0, "Distributor: Not registered to claim ERC20 token.");
+
+        uint256 tokenAmountClaimable = claimableToken(account, token);
+
+        // Transfer ERC20 via transfer function from SafeERC20
+        require(token.balanceOf(address(this)) >= tokenAmountClaimable, "Distributor: Insufficient ERC20 token balance.");
+        
+        _totalTokenDistributed[token] += tokenAmountClaimable;
+        _distributedTokens[token][account] += tokenAmountClaimable;
+        SafeERC20.safeTransfer(token, account, tokenAmountClaimable);
+
+        emit TokenFundClaimed(account, token, tokenAmountClaimable);
+    }
+
+    // @Desc: Convenient One-Time Function call for owner to distribute ether from contract to all registered payees
+    function distributeDemAllETH() external onlyOwner {
+        uint totalPayees = numRegisteredForETH();
+        for(uint256 i = 0; i < totalPayees;){
+            if(_sharesForETH[_payeesForETH[i]] > 0) {
+                payoutETH(payable(_payeesForETH[i]));
+            }
+            unchecked {i++; }
+        }
+    }
+
+       // @Desc: Convenient One-Time Function call for owner to distribute a specified ERC20 Token from contract to all registered payees
+    function distributeDemAllToken(IERC20 token_) external onlyOwner {
+        IERC20 token = token_;
+        uint totalPayees = numRegisteredForToken(token);
+        for(uint256 i = 0; i < totalPayees;){
+            payoutToken(_payeesForToken[token][i], token);
+            unchecked {i++; }
+        }
+    }
+
+      /*///////////////////////////////////////////////////////////////
+                            Internal/Private Functions
+    //////////////////////////////////////////////////////////////*/
+
+    // @Desc: To register accounts as payees for token distribution.
+    // @Note: tokenAddress_ to specify as Null Address to register a payee with shares for Ether.
+    function _registerPayeesForETH(address account_, uint256 share_) private {
+        address account = account_;
+        uint share = share_;
+        require(account != address(0), "Distributor: Null address not allowed.");
+        require(share > 0, "Distributor: Invalid shares specified.");
+        require(_sharesForETH[account_] == 0, "Distributor: Account registered for ETH distribution.");
+
+        // Save account as a payee in the mapping
+        _payeesForETH[_numRegisteredForETH++] = account_; 
+        _sharesForETH[account_] = share_;
+         _totalETHShares += share_;
+         
+        emit RegisterPayeeForETH(account, share);
+    }
```

Comment:
can you update a share?

Reply:
Nope, I thought of implementing that at first but then I felt maybe it would be better if the proportion of shares were fixed when launched to preserve perpetuity of relative shares for each payee. Yeah, it would be bad if someone made a mistake in the shares breakdown and there would be many who would prefer to have it amendable also. So yeah I just didn't implement it first haha but its needed do I will gladly do it 👍